### ISSUE TITLE
Can't change the theme on the docs site

### ISSUE THREAD

**Issue by ronanru** [trpc/trpc#2913]
### Area of Improvement

The trpc.io website uses light / dark theme based on the device theme, but if you toggle the theme with the button in the top bar and refresh the page it reverts back to the default theme. There is a "theme" record in localStorage, but the site doesn't use it when loading.

Removing the "docusaurus.preferredTheme.js" file and it's mention in "docusaurus.config.js" fixed the problem.
The other solution is to read the localStorage record in the "docusaurus.preferredTheme.js" file and skip the device theme check if it's valid. 

### 👨‍👧‍👦 Contributing

- [ ] 🙋‍♂️ Yes, I'd be down to file a PR implementing the suggested changes!

**Comment by KATT** [comment]
cc @trpc/website-docs

**Comment by juliusmarminge** [comment]
This is a "problem" that I hoped would be fixed upstream but I made that script in the interim. Essentially Docusaurus stops caring about your system preferences as soon as you press that theme-toggle, which isn't ideal if you ask me.

But as you point out, the theme shouldn't change on refresh, that's bad UX. I'm not sure what the best behavior is, but I want some way to let me system preferences decide even if I once pressed the theme toggle to test it out, without having to go into the dev tools and remove the localStorage entry?

What I did in that PR was checking if the system preferences had changed since you last visited, and if so use that, else use the localStorage one?
